src/BaseElement.js

```diff
@@ -189,7 +196,7 @@ class BaseElement extends HTMLElement {
 	 * Will trigger update() when a property was changed
 	 */
 	defineProperties() {
-		Object.keys(this.properties()).forEach((prop) => {
+		Object.keys({ ...this.injectProperties(), ...this.properties() }).forEach((prop) => {
```

Comment:
@quarkus does the order matter here? I mean you said it was not allowed to have the same property in both maps - what what if someone did it anyways (because yolo). Would it break?

Reply:
nope .. thats why the order matters ;) .
In this implemetation its actually ok to have it in both maps but due to the order proerty defaults from the property map.
If its missing in property map this will autodefine a prop for the current key.